app/code/Magento/PageCache/Model/Varnish/VclGenerator.php

```diff
@@ -228,4 +239,18 @@ private function getDesignExceptions()
     {
         return $this->designExceptions;
     }
+
+    /**
+     * Check if document root contains pub
+     *
+     * @return string
+     */
+    private function getHealthCheck() : string
+    {
+        if (strpos($this->url->getBaseUrl(), 'pub') === false) {
```

Comment:
@torhoehn missed that during the review, sorry. If base url contains `pub` - health check should also contain `pub` and vice-versa
```suggestion
        if (strpos($this->url->getBaseUrl(), 'pub') !== false) {
```

Reply:
Changed it.